sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/ServiceAPITest.groovy

```diff
@@ -197,4 +209,290 @@ class ServiceAPITest extends APISpec {
         response.getHeaders().getValue("x-ms-version") == "2019-02-02"
     }
 
+    def "Restore file system"() {
+        given:
+        def cc1 = primaryDataLakeServiceClient.getFileSystemClient(generateFileSystemName())
+        cc1.create()
+        def blobName = generatePathName()
+        cc1.getFileClient(blobName).upload(defaultInputStream.get(), 7)
+        cc1.delete()
+        def blobContainerItem = primaryDataLakeServiceClient.listFileSystems(
+            new ListFileSystemsOptions()
+                .setPrefix(cc1.getFileSystemName())
+                .setDetails(new FileSystemListDetails().setRetrieveDeleted(true)),
+            null).first()
+
+        if (!playbackMode()) {
+            Thread.sleep(30000)
+        }
+
+        when:
+        def restoredContainerClient = primaryDataLakeServiceClient
+            .undeleteFileSystem(blobContainerItem.getName(), blobContainerItem.getVersion())
+
+        then:
+        restoredContainerClient.listPaths().size() == 1
+        restoredContainerClient.listPaths().first().getName() == blobName
+    }
+
+    def "Restore file system into other file system"() {
+        given:
+        def cc1 = primaryDataLakeServiceClient.getFileSystemClient(generateFileSystemName())
+        cc1.create()
+        def blobName = generatePathName()
+        cc1.getFileClient(blobName).upload(defaultInputStream.get(), 7)
+        cc1.delete()
+        def blobContainerItem = primaryDataLakeServiceClient.listFileSystems(
+            new ListFileSystemsOptions()
+                .setPrefix(cc1.getFileSystemName())
+                .setDetails(new FileSystemListDetails().setRetrieveDeleted(true)),
+            null).first()
+
+        if (!playbackMode()) {
+            Thread.sleep(30000)
+        }
+
+        when:
+        def restoredContainerClient = primaryDataLakeServiceClient.undeleteFileSystemWithResponse(
+            new FileSystemUndeleteOptions(blobContainerItem.getName(), blobContainerItem.getVersion())
+                .setDestinationFileSystemName(generateFileSystemName()), null, Context.NONE)
+            .getValue()
+
+        then:
+        restoredContainerClient.listPaths().size() == 1
```

Comment:
Validate that the restored  container has the same name as the one passed in

Reply:
Done